src/Model/ImageLoading.py

```diff
@@ -607,25 +649,80 @@ def calculate_matrix(img_ds):
     # Equation C.7.6.2.1-1.
     # https://dicom.innolitics.com/ciods/rt-structure-set/roi-contour/30060039/30060040/30060050
     matrix_m = np.matrix(
-        [[orientation[0] * dist_row, orientation[3] * dist_col, 0,
-          position[0]],
-         [orientation[1] * dist_row, orientation[4] * dist_col, 0,
-          position[1]],
-         [orientation[2] * dist_row, orientation[5] * dist_col, 0,
-          position[2]],
-         [0, 0, 0, 1]]
+        [
+            [orientation[0] * dist_row, orientation[3] * dist_col, 0, position[0]],
+            [orientation[1] * dist_row, orientation[4] * dist_col, 0, position[1]],
+            [orientation[2] * dist_row, orientation[5] * dist_col, 0, position[2]],
+            [0, 0, 0, 1],
+        ]
     )
-    x = []
-    y = []
-    for i in range(0, img_ds.Columns):
-        i_mat = matrix_m * np.matrix([[i], [0], [0], [1]])
-        x.append(float(i_mat[0]))
 
-    for j in range(0, img_ds.Rows):
-        j_mat = matrix_m * np.matrix([[0], [j], [0], [1]])
-        y.append(float(j_mat[1]))
+    # Optimization attempt: Pre-allocate arrays
```

Comment:
A variety of attempts at optimisation.
I left them in, commented, so that they can be revisited for benchmarking.
I did this before realising that the delay was not in the "pixlut" computation (the dialog showing progress didn't indicate when pixlut processing was complete, so to outward appearance, that's where the time was being spent).